### PR TITLE
Add password max length check

### PR DESCRIPTION
**Pull request by gfiorav**
CR please, @ethervoid

### CODE REVIEW

File: app/models/carto/user.rb [hunk 1]
```diff
@@ -64,7 +66,9 @@ def name_or_username
   end
 
   def password=(value)
-    return if !value.nil? && value.length < MIN_PASSWORD_LENGTH
+    return if value.nil?
```
Comment (ethervoid) [inline app/models/carto/user.rb]: This is not true. You have to merge all the condition if not, this is going to be an OR clause

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -407,6 +407,23 @@ def get_user_by_id(user_id)
     user.destroy
   end
 
+  it "should validate password presence and length" do
```
Comment (ethervoid) [inline spec/models/user_spec.rb]: Same check in [the carto model too](https://github.com/CartoDB/cartodb/blob/bb68-add-password-max-length/spec/models/carto/user_spec.rb)

File: app/models/user.rb [hunk 3]
```diff
@@ -372,9 +373,16 @@ def validate_password_change
     if @new_password != @new_password_confirmation
       errors.add(:new_password, "New password and confirm password are not the same")
     end
-    errors.add(:new_password, "Missing new password") if @new_password.nil?
-    if !@new_password.nil? && @new_password.length < MIN_PASSWORD_LENGTH
-      errors.add(:new_password, "New password is too short (6 chars min)")
+
```
Comment (ethervoid) [inline app/models/user.rb]: I should refactor this and the `def password=` logic to one method due is the same logic

**Comment by ethervoid** [comment]
Check the things I've pointed before merge it

File: app/models/carto/user.rb (hunk 1 above)
Comment (gfiorav) [inline app/models/carto/user.rb]: yeah, tests are failing, I'll bump you when they actually pass, sorry

Thanks for pointing this out

File: spec/models/user_spec.rb (hunk 2 above)
Comment (gfiorav) [inline spec/models/user_spec.rb]: Will do

**Comment by luisbosque** [comment]
retest this please

**Comment by gfiorav** [comment]
See if you like it now, @ethervoid !

Thanks!

File: app/models/carto/user.rb [hunk 4]
```diff
@@ -65,6 +67,7 @@ def name_or_username
 
   def password=(value)
     return if !value.nil? && value.length < MIN_PASSWORD_LENGTH
+    return if !value.nil? && value.length >= MAX_PASSWORD_LENGTH
```
Comment (ethervoid) [inline app/models/carto/user.rb]: Still wrong. If the value.length is greater than MIN_PASSWORD_LENGTH is going to return and don't check the MAX_PASSWORD_LENGTH

File: app/models/user.rb [hunk 5]
```diff
@@ -458,13 +473,10 @@ def self.make_token
   end
 
   def password=(value)
-    if !value.nil? && value.length < MIN_PASSWORD_LENGTH
-      errors.add(:password, "must be at least #{MIN_PASSWORD_LENGTH} characters long")
-      return
-    end
+    return if !value.nil? && !valid_password?(:password, value, value)
```
Comment (ethervoid) [inline app/models/user.rb]: This should be if !value.nil? && valid_password? isn't it?

File: app/models/carto/user.rb (hunk 4 above)
Comment (gfiorav) [inline app/models/carto/user.rb]: They will never happen at the same time though. A password cannot be shorter than the min and longer than the max at the same time. Will refactor with an OR if you insist but there's not much of a point IMHO.

File: app/models/user.rb (hunk 5 above)
Comment (gfiorav) [inline app/models/user.rb]: No, I want to return without doing anything if the password is invalid. To check the password is invalid the password cannot be nil.

File: app/models/carto/user.rb (hunk 4 above)
Comment (ethervoid) [inline app/models/carto/user.rb]: Ok I see your point. Let it be

**Comment by ethervoid** [comment]
:+1:

**Pull request merged**